src/onnxruntime.cc

```diff
@@ -682,10 +682,40 @@ ModelState::AutoCompleteConfig()
   OrtAllocator* default_allocator;
   std::string model_path;
   {
+    TRITONSERVER_InstanceGroupKind kind;
+
+    triton::common::TritonJson::Value instance_group;
+    ModelConfig().Find("instance_group", &instance_group);
+
+    // Earlier in the model lifecycle, device checks for the instance group
+    // have already occurred. If at least one instance group with
+    // "kind" = "KIND_GPU" then allow model to use GPU else autocomplete to CPU
+    bool found_gpu_instance = false;
```

Comment:
Can be removed. Refer to earlier suggestion.

Reply:
We still have to search through the config's instances since the root of the issue is the shared lib being compiled with TRITON_ENABLE_GPU but no device was found during runtime.